### ISSUE TITLE
example for 3d numpy array registration?

### ISSUE THREAD

**Issue by coldgemini** [jeanfeydy/geomloss#3]
do you have such example? 
Thanks.

**Comment by jeanfeydy** [comment]
Hi @coldgemini ,

All the examples in [the gallery](https://www.kernel-operations.io/geomloss/_auto_examples/index.html) **work fine in 3D**: the tutorials are written the way they are because it is easier for me to show what is happening with 2D toy examples, but there really isn't any technical limitation as far as GeomLoss is concerned. See for instance the [tutorial on color transfer](https://www.kernel-operations.io/geomloss/_auto_examples/optimal_transport/plot_optimal_transport_color.html), which is a simple 3D point cloud registration in the unit RGB cube.
Going further, I will implement higher-level loss functions over the summer (such as [varifold](http://openaccess.thecvf.com/content_cvpr_2017/html/Kaltenmark_A_General_Framework_CVPR_2017_paper.html) kernel and OT losses for 3D meshes), and write self-contained rigid or spline registration examples. Before working on it, I just have to release [KeOps v1.1](https://www.kernel-operations.io/keops/index.html) first!

Note, however, that unlike KeOps, **GeomLoss will not support raw numpy input**: the native support of automatic differentiation by [PyTorch](https://pytorch.org/) is a key feature that allowed me to considerably trim down the GeomLoss interface. It really should be favored over numpy for any kind of intensive application, and is on the verge of becoming a new standard for [computational anatomy](https://en.wikipedia.org/wiki/Computational_anatomy) - my main motivation. 

Best regards,
Jean

**Comment by coldgemini** [comment]
thank you.  I'll try out. 


 
---Original---

**Comment by jeanfeydy** [comment]
You're welcome! Feel free to create an other issue if you have any other question.

**Comment by coldgemini** [comment]
did you integrate geomloss into your lddmm registration project?
